eslint.config.mjs

```diff
@@ -0,0 +1,312 @@
+    files: [`**/*${EXTENSION_TEST_FILE}`, 'packages/mui-codemod/testUtils/**/*'],
+    extends: createTestConfig({
+      useMocha: true,
+    }),
+    rules: {
+      // Disabled temporarily. Enable one by one.
+      'testing-library/prefer-screen-queries': 'off',
+      'testing-library/no-container': 'off',
+      'testing-library/no-dom-import': 'off',
+      'testing-library/no-node-access': 'off',
+      'testing-library/render-result-naming-convention': 'off',
+      'testing-library/no-await-sync-queries': 'off',
+      'testing-library/no-unnecessary-act': 'off',
+      'testing-library/no-wait-for-multiple-assertions': 'off',
+      'testing-library/no-render-in-lifecycle': 'off',
+    },
+  },
+  // Test end
+  // Docs start
+  {
+    files: ['docs/**/*'],
+    extends: createDocsConfig(),
+    rules: {
+      '@next/next/no-img-element': 'off',
+      'no-restricted-imports': [
+        'error',
+        {
+          paths: NO_RESTRICTED_IMPORTS_PATHS_TOP_LEVEL_PACKAGES,
+          patterns: NO_RESTRICTED_IMPORTS_PATTERNS_DEEPLY_NESTED,
+        },
+      ],
+    },
+  },
+  // Moved from docs/data/material/components/.eslintrc.js
+  {
+    files: ['docs/data/material/components/**/*'],
+    rules: {
+      // useful for interactions feedback
+      'no-console': ['off', { allow: ['info'] }],
+      // not very friendly to prop forwarding
+      'react/jsx-handler-names': 'off',
+    },
+  },
+  // demos
+  {
+    files: ['docs/src/pages/**/*', 'docs/data/**/*'],
+    rules: {
+      // This most often reports data that is defined after the component definition.
+      // This is safe to do and helps readability of the demo code since the data is mostly irrelevant.
+      '@typescript-eslint/no-use-before-define': 'off',
+      'react/prop-types': 'off',
+      'no-alert': 'off',
+      'no-console': 'off',
+    },
+  },
+  // Next.js entry points pages
+  {
+    files: ['docs/pages/**/*', 'packages/*/src/**/*.tsx'],
+    ignores: ['**/*.spec.tsx'],
+    rules: {
+      'react/prop-types': 'off',
+    },
+  },
+  {
+    files: ['docs/data/**/*'],
+    ignores: [
+      // filenames/match-exported sees filename as 'file-name.d'
+      // Plugin looks unmaintain, find alternative? (e.g. eslint-plugin-project-structure)
+      '**/*.d.ts',
+      'docs/data/joy/getting-started/templates/**/*',
+      'docs/data/**/{css,system,tailwind}/*',
+    ],
+    plugins: {
+      'consistent-default-export-name': eslintPluginConsistentName,
+    },
+    rules: {
+      'consistent-default-export-name/default-export-match-filename': ['error'],
+    },
+  },
+  {
+    files: ['docs/data/material/getting-started/templates/**/*'],
+    rules: {
+      // So we can use # to improve the page UX
+      // and so developer get eslint warning to remind them to fix the links
+      'jsx-a11y/anchor-is-valid': 'off',
+    },
+  },
+  // Docs end
+  {
+    files: ['**/*.d.ts'],
+    rules: {
+      'import/export': 'off', // Not sure why it doesn't work
```

Comment:
> Not sure why it doesn't work

Manual authoring of js + d.ts produced an unstable setup. We should migrate to ts really, I keep running into edge-cases caused by this time and time again.